Plugins/Creature/Creature.cpp

```diff
@@ -451,7 +453,83 @@ ArgumentStack Creature::SetAbilityScore(ArgumentStack&& args)
                 pCreature->m_pStats->SetCHABase(static_cast<uint8_t>(value));
                 break;
             default:
-                GetServices()->m_log->Notice("Calling NWNX_Creature_SetAbilityScore with invalid ability ID:%d", ability);
+                GetServices()->m_log->Notice("Calling NWNX_Creature_SetRawAbilityScore with invalid ability ID:%d", ability);
+                assert(0);
+                break;
+        }
+    }
+    return stack;
+}
+
+ArgumentStack Creature::GetRawAbilityScore(ArgumentStack&& args)
+{
+    ArgumentStack stack;
+    int32_t retval = -1;
+
+    if (auto *pCreature = creature(args))
+    {
+        const auto ability = Services::Events::ExtractArgument<int32_t>(args);
+
+        switch (ability)
+        {
+            case Constants::ABILITY_STRENGTH:
+                retval = pCreature->m_pStats->m_nStrengthBase;
+                break;
+            case Constants::ABILITY_DEXTERITY:
+                retval = pCreature->m_pStats->m_nDexterityBase;
+                break;
+            case Constants::ABILITY_CONSTITUTION:
+                retval = pCreature->m_pStats->m_nConstitutionBase;
+                break;
+            case Constants::ABILITY_INTELLIGENCE:
+                retval = pCreature->m_pStats->m_nIntelligenceBase;
+                break;
+            case Constants::ABILITY_WISDOM:
+                retval = pCreature->m_pStats->m_nWisdomBase;
+                break;
+            case Constants::ABILITY_CHARISMA:
+                retval = pCreature->m_pStats->m_nCharismaBase;
+                break;
+            default:
+                GetServices()->m_log->Notice("Calling NWNX_Creature_GetRawAbilityScore with invalid ability ID:%d", ability);
+                assert(0);
+                break;
+        }
+    }
+    Services::Events::InsertArgument(stack, retval);
+    return stack;
+}
+
+ArgumentStack Creature::ModifyRawAbilityScore(ArgumentStack&& args)
+{
+    ArgumentStack stack;
+    if (auto *pCreature = creature(args))
+    {
+        const auto ability = Services::Events::ExtractArgument<int32_t>(args);
+        const auto offset  = Services::Events::ExtractArgument<int32_t>(args); assert(value <= 255);
```

Comment:
I guess technically speaking you want to assert that

offset >= -base (underflow)
base + offset <= 255 (overflow) or, practically speaking, base + offset <= 127

When they do break it's going to be obvious enough to whomever uses it why they broke. So not a big deal really. Just something you could improve if you felt like it, or not really.